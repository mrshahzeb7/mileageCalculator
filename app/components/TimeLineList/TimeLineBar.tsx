import React from 'react';
import {StyleSheet, View} from 'react-native';
import {colors} from '../../theme';
import {responsiveHeight, responsiveWidth} from '../../utils';

interface Props {}

export const TimeLineBar: React.FC<Props> = () => {
  return <View style={styles.line} />;
};

// styles
const styles = StyleSheet.create({
  line: {
    position: 'absolute',
    left: responsiveWidth(7),
    backgroundColor: colors.primaryButton,
    width: responsiveWidth(1),
    height: responsiveHeight(100),
  },
});
