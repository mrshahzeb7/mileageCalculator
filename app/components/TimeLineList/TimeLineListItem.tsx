import React from 'react';
import {StyleSheet, Text, TextStyle, View, ViewStyle} from 'react-native';
import {icons} from '../../assets';
import {colors, iconSizes, spacing} from '../../theme';
import {Icon} from '../Icon/Icon';
import {responsiveHeight} from '../../utils';
import {TimeLineSectionItem, mileageUnit} from './timeLineListContent';

interface Props {
  item: TimeLineSectionItem;
  index: number;
}

export const timeLineListItemHeight = responsiveHeight(13);

export const TimeLineListItem: React.FC<Props> = ({item}) => {
  return (
    <View style={styles.container}>
      <View style={styles.leftWrapper}>
        <LeftIcon />
        <View style={styles.detailWrapper}>
          <Text style={styles.title}>Refueling</Text>
          <Text style={styles.date}>{item?.date}</Text>

          <Mileage
            value={item?.meterReading + ' ' + mileageUnit.milesPerLitre}
          />
        </View>
      </View>

      <Text style={styles.price}>$ {item?.cost}</Text>
    </View>
  );
};

const Mileage = ({value}: {value?: string}) => {
  return (
    <View style={styles.mileageContainer}>
      <Icon icon={icons.Meter} color={colors.textDim} size={iconSizes.tiny} />
      <Text style={styles.mileage}>{value}</Text>
    </View>
  );
};

const LeftIcon = () => {
  return (
    <Icon
      containerStyle={styles.iconContainer}
      size={iconSizes.tiny}
      icon={icons.Gas}
      color={colors.light}
    />
  );
};
const $row: ViewStyle = {
  flexDirection: 'row',
  alignItems: 'center',
};

const $textStyle: TextStyle = {
  color: colors.text,
};

const $dimTextStyle: TextStyle = {
  color: colors.textDim,
};
// styles

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: spacing.horizontal.small,
    justifyContent: 'space-between',
    height: timeLineListItemHeight,
    ...$row,
  },

  leftWrapper: {
    ...$row,
  },

  iconContainer: {
    backgroundColor: colors.primaryButton,
    height: iconSizes.large,
    width: iconSizes.large,
    borderRadius: iconSizes.large,
    alignItems: 'center',
    justifyContent: 'center',
  },
  detailWrapper: {
    marginLeft: spacing.horizontal.medium,
    marginTop: spacing.vertical.medium,
  },
  mileageContainer: {
    justifyContent: 'center',
    ...$row,
  },
  title: {
    ...$textStyle,
  },
  date: {
    marginVertical: spacing.vertical.tiny,
    ...$dimTextStyle,
  },
  mileage: {
    color: colors.textDim,
    marginLeft: spacing.horizontal.small,
    ...$dimTextStyle,
  },
  price: {
    ...$textStyle,
  },
});
