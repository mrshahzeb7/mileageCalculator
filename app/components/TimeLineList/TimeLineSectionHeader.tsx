import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {colors, spacing} from '../../theme';
import {responsiveWidth} from '../../utils';
import {Dot} from '../Dot/Dot';

interface Props {
  title: string;
}

export const TimeLineSectionHeader: React.FC<Props> = ({title}) => {
  return (
    <View style={styles.container}>
      <Dot />
      <Text style={styles.date}>{title}</Text>
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: spacing.vertical.medium,
    marginBottom: spacing.vertical.small,
  },
  date: {
    color: colors.primaryButton,
    marginLeft: responsiveWidth(8),
  },
});
