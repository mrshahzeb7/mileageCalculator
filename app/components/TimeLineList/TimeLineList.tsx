import React from 'react';
import {RefreshControl, StyleSheet, View} from 'react-native';
import {colors, spacing, styleGlobals} from '../../theme';
import {List} from '../../thirdParties';
import {keyExtractor} from '../../utils';
import {Dot} from '../Dot/Dot';
import {TimeLineBar} from './TimeLineBar';
import {TimeLineListItem, timeLineListItemHeight} from './TimeLineListItem';
import {TimeLineSectionHeader} from './TimeLineSectionHeader';
import {TimeLineContentType} from './timeLineListContent';

interface Props {
  data: TimeLineContentType;
}

export const TimeLineList: React.FC<Props> = ({data}) => {
  const onRefresh = () => {};

  return (
    <View style={styles.container}>
      <TimeLineBar />

      <List
        estimatedItemSize={timeLineListItemHeight}
        ListHeaderComponent={<Dot style={styles.dot} />}
        refreshControl={
          <RefreshControl
            tintColor={colors.light}
            refreshing={false}
            onRefresh={onRefresh}
          />
        }
        data={data}
        contentContainerStyle={styles.contentContainerStyle}
        renderItem={renderItem}
        keyExtractor={keyExtractor}
      />
    </View>
  );
};

const renderItem = (props: any) => {
  if (typeof props.item === 'string') {
    return <TimeLineSectionHeader title={props.item} />;
  }
  return <TimeLineListItem {...props} />;
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  contentContainerStyle: {
    ...styleGlobals.listPaddingBottom,
  },
  dot: {
    marginBottom: spacing.vertical.small,
  },
});
