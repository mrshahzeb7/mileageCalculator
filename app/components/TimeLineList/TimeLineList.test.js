import 'react-native';
import React from 'react';
import {TimeLineList} from './TimeLineList';
import renderer from 'react-test-renderer';

describe('TimeLineList', () => {
  it(' TimeLineList renders correctly', () => {
    const tree = renderer.create(<TimeLineList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
