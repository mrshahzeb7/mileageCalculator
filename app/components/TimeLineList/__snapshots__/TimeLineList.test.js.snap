// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineList  TimeLineList renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#93b2fd",
        "height": 1334,
        "left": 52.50000000000001,
        "position": "absolute",
        "width": 7.5,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={
        <Dot
          style={
            {
              "marginBottom": 26.68,
            }
          }
        />
      }
      applyWindowCorrection={[Function]}
      canChangeSize={true}
      contentContainerStyle={
        {
          "backgroundColor": undefined,
          "minHeight": 1,
          "minWidth": 1,
          "paddingLeft": 0,
          "paddingRight": 0,
        }
      }
      contentHeight={0}
      contentWidth={0}
      data={[]}
      dataProvider={
        DataProvider {
          "_data": [],
          "_firstIndexToProcess": 0,
          "_hasStableIds": true,
          "_requiresDataChangeHandling": false,
          "_size": 0,
          "getStableId": [Function],
          "rowHasChanged": [Function],
        }
      }
      disableRecycling={false}
      estimatedItemSize={173.42000000000002}
      extendedState={{}}
      externalScrollView={[Function]}
      finalRenderAheadOffset={250}
      forceNonDeterministicRendering={true}
      horizontal={false}
      initialOffset={0}
      initialRenderIndex={0}
      isHorizontal={false}
      keyExtractor={[Function]}
      layoutProvider={
        GridLayoutProviderWithProps {
          "_acceptableRelayoutDelta": 1,
          "_getHeightOrWidth": [Function],
          "_getLayoutTypeForIndex": [Function],
          "_getSpan": [Function],
          "_hasExpired": false,
          "_maxSpan": 1,
          "_setLayoutForType": [Function],
          "_tempDim": {
            "height": 0,
            "width": 0,
          },
          "averageWindow": AverageWindow {
            "currentAverage": 173.42000000000002,
            "currentCount": 1,
            "inputValues": [
              173.42000000000002,
            ],
            "nextIndex": 1,
          },
          "defaultEstimatedItemSize": 100,
          "layoutObject": {
            "size": undefined,
            "span": undefined,
          },
          "props": {
            "ListHeaderComponent": <Dot
              style={
                {
                  "marginBottom": 26.68,
                }
              }
            />,
            "contentContainerStyle": {
              "paddingBottom": 200.1,
            },
            "data": [],
            "estimatedItemSize": 173.42000000000002,
            "keyExtractor": [Function],
            "numColumns": 1,
            "onEndReachedThreshold": 0.2,
            "refreshControl": <RefreshControlMock
              onRefresh={[Function]}
              refreshing={false}
              tintColor="#ffffff"
            />,
            "renderItem": [Function],
            "showsVerticalScrollIndicator": false,
          },
          "renderWindowInsets": {
            "height": 0,
            "width": 0,
          },
          "shouldRefreshWithAnchoring": true,
        }
      }
      maxRenderAhead={750}
      numColumns={1}
      onEndReached={[Function]}
      onEndReachedThreshold={0}
      onEndReachedThresholdRelative={0.2}
      onItemLayout={[Function]}
      onLayout={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onSizeChanged={[Function]}
      onVisibleIndicesChanged={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
          tintColor="#ffffff"
        />
      }
      removeClippedSubviews={false}
      renderAheadOffset={0}
      renderAheadStep={250}
      renderContentContainer={[Function]}
      renderItem={[Function]}
      renderItemContainer={[Function]}
      rowRenderer={[Function]}
      scrollEventThrottle={16}
      scrollThrottle={16}
      scrollViewProps={
        {
          "contentContainerStyle": {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          "onLayout": [Function],
          "onScrollBeginDrag": [Function],
          "refreshControl": <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
            tintColor="#ffffff"
          />,
          "style": {
            "minHeight": 1,
            "minWidth": 1,
          },
        }
      }
      showsVerticalScrollIndicator={false}
      style={
        {
          "minHeight": 1,
          "minWidth": 1,
        }
      }
      suppressBoundedSizeException={true}
      windowCorrectionConfig={
        {
          "applyToInitialOffset": true,
          "applyToItemScroll": true,
          "value": {
            "endCorrection": 0,
            "startCorrection": 0,
            "windowShift": -1,
          },
        }
      }
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              {
                "paddingLeft": 0,
                "paddingTop": 0,
              }
            }
          />
          <View
            style={
              [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#93b2fd",
                    "borderRadius": 15.297966458904849,
                    "height": 30.595932917809698,
                    "marginLeft": 41.25,
                    "width": 30.595932917809698,
                  },
                  {
                    "marginBottom": 26.68,
                  },
                ]
              }
            />
          </View>
          <AutoLayoutView
            enableInstrumentation={false}
            horizontal={false}
            onBlankAreaEvent={[Function]}
            onLayout={[Function]}
            renderAheadOffset={0}
            scrollOffset={0}
            style={
              {
                "height": 0,
                "width": 0,
              }
            }
            windowSize={0}
          />
          <CellContainer
            index={-1}
            style={
              [
                undefined,
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "paddingBottom": 200.1,
                "paddingRight": 0,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
